packages/celotool/src/lib/k8s-fullnode/aws.ts

```diff
@@ -0,0 +1,77 @@
+import { AWSClusterConfig } from '../k8s-cluster/aws'
+import { BaseFullNodeDeploymentConfig } from './base'
+import { BaseNodePortFullNodeDeployer } from './base-nodeport'
+
+export interface AWSFullNodeDeploymentConfig extends BaseFullNodeDeploymentConfig {
+  clusterConfig: AWSClusterConfig
+}
+
+/**
+ * At the moment, there is no way to use a Network Load Balancer on EKS
+ * with ingress TCP & UDP traffic on the same port. Instead, we use NodePort
+ * services.
+ */
+export class AWSFullNodeDeployer extends BaseNodePortFullNodeDeployer {
+  /**
+   * Gets AWS-specific helm parameters.
+   */
+  async additionalHelmParameters() {
+    return [
+      ...(await super.additionalHelmParameters()),
+      `--set aws=true`,
+      `--set storage.storageClass=gp2`,
+    ]
+  }
+
+  /**
+   * Prints action required to remove the node ports, and removes the chart
+   */
+  async removeChart() {
+    const serviceForEachFullNode = await this.getServiceForEachFullNode()
+    const serviceNodePortsSet = serviceForEachFullNode.reduce((set: Set<number>, service: any) => {
+      // If there is no service for a full node, it is undefined. Just ignore
+      if (!service) {
+        return set
+      }
+      for (const portSpec of service.spec.ports) {
+        if (portSpec.nodePort) {
+          set.add(portSpec.nodePort)
+        }
+      }
+      return set
+    }, new Set<number>())
+    this.printNodePortsActionRequired(Array.from(serviceNodePortsSet), true)
+    await super.removeChart()
+  }
+
+  /**
+   * In order for NodePort to work on an EKS cluster, you need to manually
+   * allow ingress traffic to the specific ports for each VM in the cluster.
+   * This prints a message showing which ports need to be added and how to modify
+   * the security group that is common to all VMs in a cluster.
+   */
+  printNodePortsActionRequired(nodePorts: number[], destroy: boolean = false) {
```

Comment:
I am OK with not automating this and taking as a manual action, but I am curious if you think this could be automated to authorizing here.

Reply:
Tbh didn't really think of it-- it feels very automatable (if that's a word?) so I'll look into it